README-Substrait.md

```diff
@@ -0,0 +1,173 @@
+# Substrait Dialect for MLIR
+
+This project consist of building an input/output dialect in
+[MLIR](https://mlir.llvm.org/) for [Substrait](https://substrait.io/), the
+cross-language serialization format of database query plans (akin to an
+intermediate representation/IR for database queries). The immediate goal is to
+create common infrastructure that can be used to implement consumers, producers,
+optimizers, and transpilers of Substrait; the more transcending goal is to study
+the viability of using MLIR to implement database query compilers.
+
+## Motivation
+
+Substrait defines a serialization format for data-intensive compute operations
+similar to relational algebra as they typically occur in database query plans
+and similar systems, i.e., the "intermediate representation" (or IR) of database
```

Comment:
It doesn't feel like an IR in that sense, more like an interop, it's a common interchange format.

Reply:
Yeah, true. Changed here and added an introduction of the term "intermediate representation (IR)" below.